testing.py

```diff
@@ -968,6 +976,161 @@ def sample_cover_representation(self, name):
             media_type="image/png", content=open(sample_cover_path).read())[0]
 
 
+class SearchClientForTesting(ExternalSearchIndex):
+    """When creating an index, limit it to a single shard and disable
+    replicas.
+
+    This makes search results more predictable.
+    """
+
+    def setup_index(self, new_index=None):
+        return super(SearchClientForTesting, self).setup_index(
+            new_index, number_of_shards=1, number_of_replicas=0
+        )
+
+
+class ExternalSearchTest(DatabaseTest):
```

Comment:
This code was previously in tests/external_search.py. It worked out because there was nowhere else we needed to run an end-to-end test using a real Elasticsearch instance. But now the real-world test of `WorkList.groups()` uses an Elasticsearch instance, so I moved this class to testing.py, making it usable from any test file.